core/asset/block.go

```diff
@@ -32,11 +32,16 @@ func (reg *Registry) indexAnnotatedAsset(ctx context.Context, a *Asset) error {
 	m := map[string]interface{}{
 		"id":               a.AssetID,
 		"alias":            a.Alias,
-		"definition":       a.Definition,
+		"raw_definition":   json.HexBytes(a.RawDefinition()),
+		"vm_version":       a.VMVersion,
 		"issuance_program": json.HexBytes(a.IssuanceProgram),
 		"tags":             a.Tags,
 		"is_local":         "no",
 	}
+	adef, err := a.Definition()
+	if err == nil {
+		m["definition"] = adef
```

Comment:
The previous behaviour as of 6c24fcc504aad981577940d1169d5ec974333b48 (when adef was in the issuance program as json string) would fail to parse non-json definitions and not index such assets. Current behaviour indexes even non-json definitions and returns to the clients both original hex-encoded raw string ("raw_definition"), and return nil for the structured form ("definition").

Note that current Cores never allow creating non-JSON definitions in the first place, so current applications would be unaffected.